### DISCUSSION TITLE
Where are my music files? - not where it says!

### DISCUSSION THREAD

**Discussion by rogercreagh** [AzuraCast/AzuraCast#6901] [Q&A] [answered]
I'm very confused. I now have an azuracast test station running on a new server. (Ubuntu 22.04)

I used the recommended installation method (docker) and all seems to be working file. I have uploaded some music files within azuracast and they are playing.

In the azuracast interface it says that the media files are in `/var/azuracast/stations/station_one/media`That seems clear enough - so I look in /var/azuracast and the stated folder simply doesn't exists. There is no stations directory. Just docker.sh and a couple of other files and no subdirectories at all.

But the media is obviously somewhere because I can hear it playing - so WTF????

I need to be able to access it to share with other applications outside azuracast including a sync service (to a nextcloud or onedrive) and for listing on another website on the same server. How to I find them?

**Comment by DocTator2281** [comment]
Hi Roger,

The media is usually stored in the docker container, and that's why it's not immediately apparent where it is physically.

I did some digging and it can be accessed locally at :

` /var/lib/docker/volumes/azuracast_station_data/_data/station_name/media`

**Comment by rogercreagh** [comment]
Oh! I wonder why  on earth it doesn't say that in azuracast - that surely is a very annoying bug that it reports the wrong location.

Although the files and subfolders are listed as belonging to the user who installed azuracast, that user can't actually see or access them without using sudo. Which is clearly not viable if wanting to access them from outside azuracast - another annoyance, if not actually a bug. 

I suppose I could add the www-data user to the sudoers but that seems like a spectacularly bad idea.

I tried adding a location to the station in the normal filespace, but azuracast reports it can't access it.  

I guess this is a consequence of using docker. Why does azuracast use such a brain dead inaccessible system? It seems highly restrictive.

Is there anyway of letting azuracast access normal filespace?
Is it possible to install azuracast without using docker?
Is the only solution to have two copies of all the media - one inside docker and one outside, and ne'er the twain shall meet. Surel there must be a better way?

Please help. Thanks

**Reply by DocTator2281** [reply]
Docker is immutable - that's the whole point... it provides a package for an application that will work in any environment that Docker is supported in, that way the maintainers don't have to worry about specific environments, just docker. 

The only thing I can think of that you can do is an Ansible install. It's not supported by the azuracast devs, but it might be better for you to be able to get at your media without it being in a docker container. 

Instructions : https://github.com/AzuraCast/ansible-installer

Keep in mind you'll want a clean server for this type of install, it won't get along with other applications on the same server.

**Comment by Moonbase59** [comment, marked as answer]
It’s just Docker. Go learn.

Actually, you can mount everything to everywhere, preferably use `/var/azuracast/docker-compose.override.yml` (on the host) for that. Here’s a simple example where I mount a complete NAS folder (mounted via NFS on the host) into my AzuraCast Docker:

```
services:
  web:
    volumes:
      - /mnt/Musik/Tagged:/var/azuracast/stations/niteradio/media/Tagged
```

Left part (before the colon) is the path on your host, right part (after the colon) the path inside your Docker (what AzuraCast sees).

**Reply by rogercreagh** [reply]
Thanks, that is exactly what I need.

**Reply by Moonbase59** [reply]
Glad I could help!
